custom_components/owlintuition/sensor.py

```diff
@@ -260,57 +268,61 @@ def __init__(self, owldata, sensor_name, sensor_type, phase=0):
         self._phase = phase
         self._owl_class = SENSOR_TYPES[sensor_type][3]
         self._state = None
-        self._attrs = {
-            ATTR_LAST_UPDATE: None,
-            ATTR_LAST_RESET: None,
-            ATTR_DEVICE_CLASS: SENSOR_TYPES[sensor_type][4],
-            ATTR_STATE_CLASS: STATE_CLASS_TOTAL_INCREASING if SENSOR_TYPES[sensor_type][4] == c.DEVICE_CLASS_ENERGY \
-                              else (STATE_CLASS_MEASUREMENT if SENSOR_TYPES[sensor_type][4] else None),
-            c.ATTR_ATTRIBUTION: 'Powered by OWL Intuition'
-            }
+        self._attr_attribution = POWERED_BY
+        self._attr_device_class = SENSOR_TYPES[self._sensor_type][4]
+        self._attr_native_unit_of_measurement = SENSOR_TYPES[self._sensor_type][1]
+        self._attr_state_class = SENSOR_TYPES[self._sensor_type][5]
 
     @property
-    def name(self):
-        """Return the name of this sensor."""
+    def _attr_name(self):
+        """Return the current name for this sensor."""
         return self._name
 
+    # @property
+    # def _attr_device_class(self):
```

Comment:
Why would you introduce those commented properties if they are statically defined in the `__init__`?

Reply:
Sorry, cleaned it up. Was left over from some experimentation.